espnet/bin/lm_train.py

```diff
@@ -62,28 +65,31 @@ def get_parser():
     parser.add_argument('--report-interval-iters', default=100, type=int,
                         help="Report interval iterations")
     # task related
-    parser.add_argument('--train-label', type=str, required=True,
+    parser.add_argument('--train-label', type=str, required=required,
                         help='Filename of train label data')
-    parser.add_argument('--valid-label', type=str, required=True,
+    parser.add_argument('--valid-label', type=str, required=required,
                         help='Filename of validation label data')
     parser.add_argument('--test-label', type=str,
                         help='Filename of test label data')
     parser.add_argument('--dump-hdf5-path', type=str, default=None,
                         help='Path to dump a preprocessed dataset as hdf5')
     # training configuration
     parser.add_argument('--opt', default='sgd', type=str,
-                        choices=['sgd', 'adam'],
                         help='Optimizer')
     parser.add_argument('--sortagrad', default=0, type=int, nargs='?',
                         help="How many epochs to use sortagrad for. 0 = deactivated, -1 = all epochs")
     parser.add_argument('--batchsize', '-b', type=int, default=300,
                         help='Number of examples in each mini-batch')
+    parser.add_argument('--accum-grad', type=int, default=1,
+                        help='Number of gradient accumueration')
     parser.add_argument('--epoch', '-e', type=int, default=20,
                         help='Number of sweeps over the dataset to train')
     parser.add_argument('--early-stop-criterion', default='validation/main/loss', type=str, nargs='?',
                         help="Value to monitor to trigger an early stopping of the training")
     parser.add_argument('--patience', default=3, type=int, nargs='?',
                         help="Number of epochs to wait without improvement before stopping the training")
+    parser.add_argument('--scalers', default=None, action="append", type=lambda kv: kv.split("="),
+                        help='optimizer schedulers, e.g., "--scalers lr=noam --lr-noam-warmup 1000".')
```

Comment:
Why the name is `scalers` but not `schedulers`?
Could you teach me what stand for `lr=`?

Reply:
Yeah, the name `schedulers` is better. Actually, you can specify any hyperparameters in optimizer like `--schedulers momentum=noam` then its hyper-hyperparameter is `--momentum-warmup 5000`

Reply:
I will add this behaviour to specify multiple scheduling in the help.